packages/vertexai/src/requests/stream-reader.ts

```diff
@@ -76,6 +78,7 @@ async function* generateResponseSequence(
       break;
     }
 
+    deleteEmptyTextParts(value);
```

Comment:
Don't we only care about the aggregated response returned in getResponsePromise(), because that's what goes into the chat history? Can we just do this in line ~169 or so where we process aggregatedResponse.candidates[i]?

Reply:
Good point- it's not really necessary to prevent empty text parts from going through the stream. I changed the behaviour so that empty text parts are only removed in the aggregated response.

`aggregateResponses` feels a little tricky to work with, so I may have implemented this incorrectly.